starter-code/src/viking.js

```diff
@@ -1,11 +1,76 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+   this.health = health;
+   this.strength = strength;
+}
+Soldier.prototype.attack = function () {
+    return this.strength;
+}
+Soldier.prototype.receiveDamage = function (damage) {
+    this.health -= damage;
+}
+//console.log(Soldier.receiveDamage(50));
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+    Soldier.call(this, health, strength);
+    this.name = name;
+}
+Viking.prototype = Object.create(Soldier.prototype);
 
+Viking.prototype.receiveDamage = function (damage) {
+    this.damage=damage;
+    this.health -= this.damage;
+    if (this.health > 0) {
+        return this.name + " has received " +  this.damage + " points of damage" ;    
+    } else {
+        return this.name +" has died in act of combat";
+    }
+}
+Viking.prototype.battleCry = function () {
+    return "Odin Owns You All!";
+} 
 // Saxon
-function Saxon() {}
-
+function Saxon(health, strength) {
+    Soldier.call(this, health, strength);
+}
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.receiveDamage = function (damage) {
+    this.damage = damage;
+    this.health -= this.damage;
+    if (this.health > 0) {
+        return "A Saxon has received " + this.damage + " points of damage";
+    } else {
+        return "A Saxon has died in combat";
+    }
+}
 // War
-function War() {}
+function War() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+}
+
+War.prototype.addViking = function(Viking) {
+    this.vikingArmy.push(Viking);
+}
+
+War.prototype.addSaxon = function(Saxon) {
+    this.saxonArmy.push(Saxon);
+}
+
+War.prototype.vikingAttack = function() {
+    //choosing a random saxon from the army
+  var indexRandomSaxon = Math.floor(Math.random()*this.saxonArmy.length);
+  var saxon = saxonArmy[indexRandomSaxon];
```

Comment:
Ici tu as la bonne méthode, il y a simplement une petite erreur de syntaxe : il faut indiquer this.saxonArmy si tu veux que ça fonctionne ;)

Un exemple de manière de faire la vikingAttack method:

```javascript
War.prototype.vikingAttack = function() {
    
    var randomSaxNum = Math.floor( Math.random() * this.saxonArmy.length);
    var randomSaxon = this.saxonArmy[randomSaxNum];
    
    var randomVikNum = Math.floor( Math.random() * this.vikingArmy.length);
    var randomViking = this.vikingArmy[randomVikNum];

    var result = randomSaxon.receiveDamage( randomViking.attack() );

    if( randomSaxon.health <= 0 ) {
        this.saxonArmy.splice( randomSaxNum, 1 );
    }

    return result;
};
```